engine/src/player-platform.cpp

```diff
@@ -2030,8 +2033,14 @@ Boolean MCPlayer::prepare(MCStringRef options)
 	if (m_platform_player == nil || m_should_recreate || !dontuseqt)
     {
         if (m_platform_player != nil)
+		{
+			// IM-2016-05-27: [[ Bug 17697 ]] Clear native layer when releasing player
+			SetNativeView(nil);
             MCPlatformPlayerRelease(m_platform_player);
+		}
```

Comment:
Are the lifetimes of the `MCPlatformPlayer` instance and the native layer the same?  I.e. are they always created and destroyed together?